HaxeManual/08-compiler-features.tex

```diff
@@ -734,3 +734,28 @@ \section{Static Analyzer}
 \item[\ic{fusion}]: Moves variable expressions to its usage in case of single-occurrence. By default, only compiler-generated variables are handled. This can be changed by using the compiler flag `\ic{-D analyzer-user-var-fusion} or the metadata \ic{@:analyzer(user_var_fusion)}.
 \item[\ic{purity_inference}]: Infers if fields are ``pure'', i.e. do not have any side-effects. This can improve the effect of the fusion module.
 \end{description}
+
+
+\section{Loop unrolling}
+\label{cr-loop-unrolling}
+\since{4.0.0}
+
+The compiler tries to unroll short iterations of \expr{for} loops over constant ranges. This can produce faster code that uses fewer variables.
+
+\begin{lstlisting}
+for (i in 0...4) {
+  test(i);
+}
+\end{lstlisting}
+
+The above code turns into something similar to:
+
+\begin{lstlisting}
+test(0);
+test(1);
+test(2);
+test(3);
+\end{lstlisting}
+
+Whether a loop is unrolled or not depends on the number of iterations of the loop and the number of expressions inside the loop body. This threshold can be configured using the \expr{-D loop_unroll_max_cost=...} compile-time define.
```

Comment:
This should mention the default value (250) so people get a rough idea of what dimension we're talking about here. Might also want to mention that we multiply the number of iterations with the number of expressions in the loop body.